applications/ShapeOptimizationApplication/python_scripts/algorithm_penalized_projection.py

```diff
@@ -66,25 +64,36 @@ def CheckApplicability(self):
             raise RuntimeError("Penalized projection algorithm requires definition of a constraint!")
         if self.constraints.size() > 1:
             raise RuntimeError("Penalized projection algorithm only supports one constraint!")
+
     # --------------------------------------------------------------------------
     def InitializeOptimizationLoop(self):
+        self.model_part_controller.ImportOptimizationModelPart()
+        self.model_part_controller.InitializeMeshController()
+
+        self.analyzer.InitializeBeforeOptimizationLoop()
+
+        self.design_surface = self.model_part_controller.GetDesignSurface()
+
+        self.mapper = mapper_factory.CreateMapper(self.design_surface, self.design_surface, self.mapper_settings)
```

Comment:
In other places in Kratos, the object (here mapper) is created in the constructor, and here only the `Initialize` function should be called. Or is there a specific reason you create the mapper in this place now?

Reply:
same for the other objects below.

Reply:
The reason is, that these objects require an imported model part. And it seemed more reasonable to import the model part in InitializeOptimizationLoop, and leave the constructor with only adding the variables. That's why also these objects are created there. Actually, the objects you mention should not just expect an imported mdpa right from the beginning. But this is a bit more effort.

Reply:
I see. Importing the model part in the Initialize function is in line with other implementations in Kratos e.g. of the analysis, so :+1: . 

Please initialize the variables with `None` in constructor. Maybe we can even add a comment to remember that we should separate some code from the constructor of those classes into an Initialize function.